### PR TITLE
Introduce CI deployment and tests through travis. 

### PR DESCRIPTION
**Pull request by cheesema**
The intention of this PR is to introduce CI deployment of PyLibAPR across linux, windows, and hopefully also OSX (no work done yet).

- [x] Linux deployment to test pypi.
- [x] Windows deployment to test pypi (working, but need to figure out if cibuildwheel can work for the different python versions)
- [x] OSX deployment to test pypi; building but image not found issues.
- [x] Tests running on push in one of the environments, linux?
- [x] Add deployment and build tags to Readme.
- [x] Deal with the case where cmake < 1.18 or add the restriction. (Cmake changed the syntax for python detection)
- [x] See if we can remove libtiff

### CODE REVIEW

**Comment by cheesema** [comment]
Gave up on getting the "clearner" way for the windows builds as it was not deceting the python versions correctly. However everything seems to be working. Once we are okay with things and setting up will need of course to change to proper pypi and also to put some logic on the deployment only hapening on tag, and only on master.

The code changes outside of the travis are small enough that you could quickly go through and review them and suggest any changes. I had remove a const in the wrappers to get the build working on my osx build, and I also had make some iterators be passed in by copy to deal with old gcc and openmp issues.

But this now builds for linux, windows, osx, across 3.6-3.9, and will run the python tests on linux and osx.

**Comment by cheesema** [comment]
If the OSX build is failing, I think we ignore it for now. As it was working for the previous days, and brew just stopped retuning the downloads.

**Comment by cheesema** [comment]
- [ ] Change the windows to be cibuildwheel if fixed.
- [ ] Maxflow apr_segmentation_demo fails. On windows this fails.

File: .gitmodules [hunk 1]
```diff
@@ -5,5 +5,5 @@
 [submodule "external/LibAPR"]
 	path = external/LibAPR
 	url = https://github.com/AdaptiveParticles/LibAPR.git
-	branch = develop_joel
+	branch = tmp_install_windows_joel
```
Comment (joeljonsson) [inline .gitmodules:8]: should remember to make this master once we've pushed through the merge in LibAPR

File: requirements.txt [hunk 2]
```diff
@@ -1,6 +1,5 @@
 cmake-setuptools
```
Comment (joeljonsson) [inline requirements.txt:1]: we no longer rely on `cmake-setuptools`
could also include tifffile here, although it's installed with other dependencies

File: CMakeLists.txt [hunk 3]
```diff
@@ -42,8 +42,14 @@ if(NOT PYAPR_PREFER_EXTERNAL_LIBAPR OR NOT APR_FOUND)
     set(APR_USE_OPENMP ${PYAPR_USE_OPENMP} CACHE BOOL "" FORCE)
     set(APR_USE_CUDA ${PYAPR_USE_CUDA} CACHE BOOL "" FORCE)
 
+    if(CMAKE_COMPILER_IS_GNUCC)
+        set(APR_PREFER_EXTERNAL_BLOSC OFF CACHE BOOL "" FORCE)
```
Comment (joeljonsson) [inline CMakeLists.txt:46]: is there an issue with using external blosc here?

**Review by joeljonsson — APPROVED** [review]
tried pip installation on osx, windows and ubuntu -- all fine (except graphcuts on windows). Ran some benchmarks and compared to manual installations with openmp off, and omp seems to be working properly

File: .gitmodules (hunk 1 above)
Comment (cheesema) [reply .gitmodules:8]: Yeah i'll update this to master.

File: requirements.txt (hunk 2 above)
Comment (cheesema) [reply requirements.txt:1]: I think it woudl be better to make it explicit. Will have a go at removing cmake-setup tools and adding tifffile.

File: CMakeLists.txt (hunk 3 above)
Comment (cheesema) [reply CMakeLists.txt:46]: Yeah there was, i'll retry it now with master. Not sure why as I was not getting any issues direclty with LibAPR

**Comment by cheesema** [comment]
apr_segmentation_demo seems to work fine for me on windows? @joeljonsson

File: .gitmodules (hunk 1 above)
Comment (cheesema) [reply .gitmodules:8]: I'm a bit confused on when this is used vs when you just add the commit update via git add? I have done both.

**Comment by cheesema** [comment]
@joeljonsson the fail to me only happens when reading in apr with float type it seems. By changing the data type read in it no longer faults. Can we through a warning or error in this case?

**Comment by cheesema** [comment]
I think I may have broken something in the raycaster, @joeljonsson does it work on your datasets? I just get stuck inside with a broken view.

**Pull request merged**